src/test/java/edu/cornell/kfs/concur/businessobjects/ValidationResultTest.java

```diff
@@ -18,54 +22,80 @@ public class ValidationResultTest {
     private static final String TEST_STRING_ONE = "test string 1";
     private static final String TEST_STRING_TWO = "test string 2";
     private static final String TEST_STRING_THREE = "test string 3";
+    private static final String TEST_STRING_FOO = "FOO";
     private static final String TEST_STRING_THREE_UPPER_CASE = StringUtils.upperCase(TEST_STRING_THREE, Locale.US);
 
-    @Before
+    @BeforeEach
     public void setUp() throws Exception {
         validationResult = new ValidationResult();
     }
 
-    @After
+    @AfterEach
     public void tearDown() throws Exception {
         validationResult = null;
     }
-
-    @Test
-    public void verifyAddMessage() {
-        validationResult.addMessage(TEST_STRING_ONE);
-        validationResult.addMessage(TEST_STRING_TWO);
-        validationResult.addMessage(TEST_STRING_ONE);
-        validationResult.addMessage(TEST_STRING_THREE);
-        validationResult.addMessage(TEST_STRING_THREE_UPPER_CASE);
-        assertEquals("There should only be three lines", 3, validationResult.getMessages().size());
+    
+    @ParameterizedTest
+    @MethodSource("addMessageParams")
+    public void testAddErrorMessage(List<String> messagesToAdd, int numberOfMessagesExpected, int baseListSize) {
+        for (String message : messagesToAdd) {
+            validationResult.addErrorMessage(message);
+        }
```

Comment:
Could you use Collections.addAll(messageList, messages) instead of the for loop?

Reply:
The addErrorMessage() function prevents duplicate messages, so we want to call it like this.  this unit test is validating that functionality is working as expected.